AWSPinpointTests/AWSPinpointTargetingClientTests.m

```diff
@@ -79,19 +80,22 @@ - (AWSPinpointConfiguration *)getDefaultAWSPinpointConfiguration {
 - (AWSPinpointConfiguration *)getAWSPinpointConfigurationWithOptOut:(BOOL)optOut {
     AWSPinpointConfiguration *configuration = [self getDefaultAWSPinpointConfiguration];
 
-    [self setApplicationLevelOptOut:configuration withOptOut:optOut];
-
-    return configuration;
-}
-
-- (void)setApplicationLevelOptOut:(AWSPinpointConfiguration *)configuration withOptOut:(BOOL)optOut {
     configuration.isApplicationLevelOptOut = ^BOOL{
         return optOut;
     };
+
+    return configuration;
 }
 
-- (void)setApplicationLevelOptOut:(BOOL)optOut {
-    [self initializePinpointWithConfiguration:[self getAWSPinpointConfigurationWithOptOut:optOut]];
+- (void)setDeviceTokenInUserDefaults {
+    const unsigned char currentTokenBytes[] = {
+        0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+        0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+    };
+    NSData *currentTokenData = [[NSData alloc] initWithBytes:currentTokenBytes length:16];
+    [self.userDefaults removeObjectForKey:AWSDeviceTokenKey];
+    [self.userDefaults setObject:currentTokenData forKey:AWSDeviceTokenKey];
```

Comment:
The unit test should have its own constant for AWSDeviceTokenKey, rather than relying on the one in production code.

Reply:
addiing